Source/WebCore/Modules/permissions/Permissions.h

```diff
@@ -36,24 +36,24 @@ class JSObject;
 
 namespace WebCore {
 
-class Navigator;
+class NavigatorBase;
 class PermissionStatus;
 
 template<typename IDLType> class DOMPromiseDeferred;
 
 class Permissions : public RefCounted<Permissions> {
     WTF_MAKE_ISO_ALLOCATED(Permissions);
 public:
-    static Ref<Permissions> create(Navigator&);
+    static Ref<Permissions> create(NavigatorBase&);
     ~Permissions();
 
-    Navigator* navigator();
+    NavigatorBase* navigator();
     void query(JSC::Strong<JSC::JSObject>, DOMPromiseDeferred<IDLInterface<PermissionStatus>>&&);
 
 private:
-    explicit Permissions(Navigator&);
-
-    WeakPtr<Navigator> m_navigator;
+    explicit Permissions(NavigatorBase&);
+    
```

Comment:
You're adding trailing spaces here :/ Ideally, this should be a blank line, without trailing spaces.

Reply:
Fixed!